model/src/w3metamd.F90

```diff
@@ -17,39 +34,45 @@ MODULE W3METAMD
 !     and a linked list construct for dynamic storage.
 !/ ------------------------------------------------------------------- /
 
-      ! Values to represent "unset" data:
+      !> Value to represent "unset" character variable
       CHARACTER(LEN=*), PARAMETER :: UNSETC = "unset"
+      !> Value to represent "unset" real variable
       REAL, PARAMETER             :: UNSETR = HUGE(1.0)
 
-      ! Type for storing a user defined metadata pair:
+      !> Type for storing a user defined metadata pair as linked list element
       TYPE META_PAIR_T
-        CHARACTER(LEN=64)  :: ATTNAME = UNSETC
-        CHARACTER(LEN=120) :: ATTVAL = UNSETC
-        CHARACTER          :: TYPE = 'c'    ! c,i,f/r
-        TYPE(META_PAIR_T), POINTER  :: NEXT
+        CHARACTER(LEN=64)  :: ATTNAME = UNSETC  !< Attribute name
+        CHARACTER(LEN=120) :: ATTVAL = UNSETC   !< Attribute value
+        CHARACTER          :: TYPE = 'c'        !< Attribute type (c,i,f/r)
+        TYPE(META_PAIR_T), POINTER  :: NEXT     !< Pointer to next node
       END TYPE META_PAIR_T
 
+      !> Linked list of meta data pairs
       TYPE META_LIST_T
         TYPE (META_PAIR_T), POINTER :: HEAD => NULL(), TAIL => NULL()
-        INTEGER                     :: N = 0
+        INTEGER                     :: N = 0    !< Num elements in list
       END TYPE META_LIST_T
 
-      ! Interface to facilitate adding real/int/character values to list
+      !> Interface to facilitate adding real/int/character values to list
       INTERFACE META_LIST_APPEND
-        MODULE PROCEDURE META_LIST_APPEND_M
-        MODULE PROCEDURE META_LIST_APPEND_R
-        MODULE PROCEDURE META_LIST_APPEND_I
-        MODULE PROCEDURE META_LIST_APPEND_C
+        MODULE PROCEDURE META_LIST_APPEND_M !< Append a META_PAIR_T
+        MODULE PROCEDURE META_LIST_APPEND_R !< Append a REAL value
+        MODULE PROCEDURE META_LIST_APPEND_I !< Append an INTEGER value
+        MODULE PROCEDURE META_LIST_APPEND_C !< Append a CHARACTER value
       END INTERFACE META_LIST_APPEND
 
-
       CONTAINS
       
 
 !/ ------------------------------------------------------------------- /
-      SUBROUTINE DEL_META_LIST(LIST)
-!     Deletes all entries in list
+!> @brief Deletes all entries in list.
```

Comment:
You do not need the brief tag for subprograms. The first sentance will automatically be used as the brief description for subroutines/functions.

Brief tags are needed for files and modules, but not subprograms.

Reply:
This has come up and I've kicked it around, though I think I like having the brief tags in even for subroutines, for two reasons.  First, it's consistent.  For getting other developers to add doxygen to their code who may not have much experience doxygen, it's going to be less confusing, which will encourage people to comply.  Second, I believe it visually adds some structure.  This way when you are quickly scanning through code it's clear that it's the brief and part of the next header, versus potentially a stray comment from whatever's above it.